### PR TITLE
【back】constant.py を constants/ ディレクトリに分割 + Choices を choice.py に集約

### PR DESCRIPTION
**Pull request by ANFINII**
## Summary
- `api/utils/constant.py` を `api/utils/constants/` パッケージに分割（`media.py` / `notification.py` / `choice.py`）
- モデル内に散らばっていた `choices` 定義を `constants/choice.py` に集約
- すべての import path を `api.utils.constant` → `api.utils.constants.xxx` に更新

### `myus/api/utils/constants/` 新設
- `__init__.py`: 新規追加
- `media.py`: 旧 `constant.py` のうちメディアモデル系の辞書（`model_dict`, `model_list`, `model_pjax`, `model_create_pjax`, `model_comment_dict`, `model_like_dict`, `model_media_comment_dict`, `model_media_comment_no_dict` など）を移管
- `notification.py`: `notification_type_no` / `notification_type_model` を分離して配置
- `choice.py`: 各 Django モデルで個別に定義していた `choices` を集約
  - `GENDER_CHOICES` / `ADVERTISE_TYPE_CHOICES`
  - `PLAN_CHOICES` / `PLAN_ACTION_CHOICES`
  - `CURRENCY_CHOICES` / `PROVIDER_CHOICES` / `PAYMENT_STATUS_CHOICES` / `SUBSCRIPTION_STATUS_CHOICES`

### モデル側のリファクタリング
- `db/models/common.py`: `Advertise.choice` を削除し `ADVERTISE_TYPE_CHOICES` を参照
- `db/models/user.py`: `Profile.gender_type` / `UserPlan.plans` / `UserPlanHistory.plans` / `UserPlanHistory.actions` をモデル内定義から `*_CHOICES` 参照へ変更
- `db/models/payment.py`: import 元を `constants.choice` に変更
- `db/models/notification.py`: 未使用の `GenericForeignKey` / `ContentType` import を削除

### import path 更新
- `api/admin.py` / `app/modules/context_data.py` / `get_form.py` / `notification.py` / `pjax.py` / `search.py` / `app/views.py` の import 元を `api.utils.constant` → `api.utils.constants.{media,notification}` に修正

### CODE REVIEW

**Review by ANFINII — COMMENTED** [review]
## 総評

`constant.py` 単一ファイルから `constants/` パッケージへの分割は責務の分離が明確で、Choices をモデル内ローカル定義から `choice.py` に集約した点も保守性が大幅に向上しています。差分も import 経路の差し替えが中心で、ロジック変更を含まない安全なリファクタリングです。以下、気づいた点を共有します。

### 良い点
- `notification_type_no` / `notification_type_model` を `notification.py` に分離したことで、メディア関連 (`media.py`) との循環 import 経路が整理されている
- 既存ブランチ `back_fix_constant_circular_import` の課題に通じる構造改善で、`db.models.notification` 側で不要だった `GenericForeignKey` / `ContentType` import まで掃除されており丁寧
- `Profile.gender_type` のようにモデルクラスのクラス変数として `choices` を定義していたものを `*_CHOICES` 定数化したことで、テーブル列以外の余計なクラス属性が消え、`makemigrations` でのノイズも減るはず

### 気になる点

1. **`PLAN_CHOICES` / `PLAN_ACTION_CHOICES` の値が enum インスタンスのまま**
   `choice.py` の以下の定義について、PR #838 で集約されている他の `CURRENCY_CHOICES` 等と同じスタイルですが、Django の `CharField(choices=...)` に渡す際は `(value, label)` の `value` を str に評価することが期待されます。
   ```python
   PLAN_CHOICES = [(p, p) for p in PlanName]
   PLAN_ACTION_CHOICES = [(a, a) for a in PlanAction]
   ```
   - `PlanName` / `PlanAction` が `str, Enum` の混在を継承していれば動作しますが、純粋な `Enum` の場合 `default=PlanName.FREE` と組み合わせて migration の `default` 値や DB 保存時にハマる可能性があります。enum の定義が `class PlanName(str, Enum)` になっているか念のため確認推奨。

2. **`choice.py` の `tuple` / `list` 形式の不統一**
   ```python
   GENDER_CHOICES = ((..., "男性"), (..., "女性"), (..., "秘密"))           # tuple of tuples
   ADVERTISE_TYPE_CHOICES = (("all", "全体"), ("one", "個別"))              # tuple of tuples
   PLAN_CHOICES = [(p, p) for p in PlanName]                                # list of tuples
   ```
   Django 的にはどちらも動きますが、`constants` として集約するならスタイルを揃えると視認性が上がります（既存 `*_CHOICES` が list 内包表記なので、固定値も list 化する or 内包表記を tuple 化するなど）。

3. **`media.py` に `choices` 系定数を残していない**
   旧 `constant.py` の責務が「メディア辞書 + 各種 CHOICES」だったため、責務分割の意図としては正しいですが、`media.py` という命名がメディアモデル専用に振り切れているので、後から「比較用辞書 (`model_dict`)」や「通知タイプ番号辞書」を追加するときに `media.py` に置くべきか迷う気がします。`model_dict` のような汎用ディスパッチ辞書だけ `dispatch.py` 等に分けるのは将来の改善余地。

4. **`Profile.message` / `Profile.phone_no` の扱い**
   `user.py` の `Profile` クラスから `gender_type` は外しましたが、`message` (`RegexValidator` のエラーメッセージ文字列) と `phone_no` (`RegexValidator` インスタンス) はクラス変数のまま残っています。`gender_type` を constants 側に移動した一貫性で言えば、ここも `constants/validator.py` 的なところに集約しても良いかもしれません。今回のスコープからは外れるので別 PR で OK。

5. **`payment.py` の import 順**
   ```python
   from api.utils.constants.choice import CURRENCY_CHOICES, PROVIDER_CHOICES, PAYMENT_STATUS_CHOICES, SUBSCRIPTION_STATUS_CHOICES
   ```
   PR #838 のスタイル（`CURRENCY → PROVIDER → SUBSCRIPTION_STATUS → PAYMENT_STATUS`）から `PAYMENT_STATUS → SUBSCRIPTION_STATUS` に並び順が変わっています。`choice.py` 内の定義順 (`PAYMENT_STATUS_CHOICES → SUBSCRIPTION_STATUS_CHOICES`) には合わせていますが、PR を追って読む人向けには意図的に並び替えた旨があると親切。

### 動作確認のお願い
- `python manage.py makemigrations --check` で choices 文字列の差分が出ないこと（Django は choices を migration の比較対象に含めるため、enum を経由した場合に空 migration が出る可能性あり）
- 開発サーバを起動し `/admin` 画面で `Advertise.type` / `Profile.gender` / `UserPlan.plan` / `UserPlanHistory.action` の選択肢が従来通り表示されること
- 通知系 (`notification_type_model` を経由するフロー) のスモークテスト

総じてリファクタリングの方向性は非常に良いと思います！上記は提案ベースなのでマージ判断は問題なしです。
